src/humancellatlas/data/metadata/helpers/date_and_time.py

```diff
@@ -0,0 +1,53 @@
+from datetime import (
+    datetime,
+    timezone,
+)
+from typing import (
+    Optional,
+)
+
+
+def datetime_from_string(s: str) -> Optional[datetime]:
+    """
+    Convert a date string into a tz-aware (UTC) datetime. The timezone can be
+    specified as an offset in '±HHMM' format, or using the timezone symbol 'Z'
+    for '+0000'.
+
+    >>> datetime_from_string('2021-05-05T21:24:26Z')
+    datetime.datetime(2021, 5, 5, 21, 24, 26, tzinfo=datetime.timezone.utc)
+
+    >>> datetime_from_string('2021-05-05T21:24:26.174274Z')
+    datetime.datetime(2021, 5, 5, 21, 24, 26, 174274, tzinfo=datetime.timezone.utc)
+
+    >>> datetime_from_string('2021-05-05T21:24:26.174274+0000')
```

Comment:
```suggestion
    >>> datetime_from_string('2021-05-05T21:24:26.174274+00:00')
```

https://datatracker.ietf.org/doc/html/rfc3339#section-5.6 requires the colon